example-workspace/example.py

```diff
@@ -0,0 +1,77 @@
+import os
+import sys
+
+from google.protobuf import text_format
+from protoc_gen_validate import validator
+from foo import bar_pb2
+
+EX_FAILURE = 1
+
+
+def main(filenames):
+    if not filenames:
+        print("No inputs provided; exiting")
+        return os.EX_OK
+
+    success_count = 0
+    for filename in filenames:
+        bars = bar_pb2.Bars()
+        try:
+            with open(filename, 'r') as fh:
+                text_format.Parse(fh.read(), bars)
+        except IOError as error:
+            print(
+                "Failed to open file '{filename}': {error}".format(
+                    filename=filename,
+                    error=error,
+                ),
+                file=sys.stderr,
+            )
+            return EX_FAILURE
```

Comment:
This is super-verbose error handling, thanks. WDYT of getting rid of the try-catch blocks here and moving them all to the __main__ block below? Any error in this loop will still throw, and then we can omit the return values since if `main()` returns without throwing that implies success.

Reply:
If we move the try-catch to the main block below then we lose the information on filename and type.

We could replace the returns with `raise RuntimeError from error` and then catch RuntimError below.